### DISCUSSION TITLE
How to animate rotation over 360 degrees?

### DISCUSSION THREAD

**Discussion by benjaminZarate** [KyryloKuzyk/PrimeTween#20] [FAQ] [answered]
Hi! I'm trying to rotate a UI element 360 degrees but it rotates to 0 or 180 

`float zAxis = indicator.localRotation.z + 360;`
`Tween.LocalRotation(indicator.transform, endValue: Quaternion.Euler(0, 0, zAxis), duration: 5);`
am I doing something wrong or I need to do something else?

**Comment by KyryloKuzyk** [comment, marked as answer]
Hi! Quaternions don't represent the Euler angles, they represent the 'orientation' of the object. Because of this, these two rotations are actually identical Quaternion.Euler(0, 0, 360) == Quaternion.Euler(0, 0, 0) and the tween will do nothing. For the same reason, it's impossible to animate rotations beyond 180 degrees using Quaternions.

To address this, there is the Tween.EulerAngles() method in PrimeTween that can animate the [eulerAngles](https://docs.unity3d.com/ScriptReference/Transform-eulerAngles.html) property. It can be used to animate loading indicator like this:

```csharp
Tween.EulerAngles(loadingIndicatorTransform, Vector3.zero, new Vector3(0, 0, 360), 1, Ease.Linear, -1);
```

Please see more info in [this post](https://forum.unity.com/threads/primetween-high-performance-animations-and-sequences.1479609/page-2#post-9336551).
